### PR TITLE
 Set cu_idx, CU start and done time in command packet

### PR DESCRIPTION
**Pull request by mamin506**
In MPSoC KDS,
1.  Record the CU index of which CU execute the command.
2.  Record timestamp just before start CU and notify user application. 
Once the command finished, user could get that above information from the exec buf.

This change is only for start hardware CU.
For configure, soft kernel command etc., this should have no impact.

### CODE REVIEW

**Comment by maxzhen** [comment]
ok to test

File: src/runtime_src/driver/zynq/drm/zocl/sched_exec.c [hunk 1]
```diff
@@ -755,6 +755,60 @@ set_cmd_state(struct sched_cmd *cmd, enum cmd_state state)
...
+ *
+ * The timestamp is reflected externally through the command packet
+ * TODO: Should have scheduler profiling solution in the future.
+ *
+ * @cmd: command object
+ * @ts: timestamp type
+ */
+inline void
+set_cmd_ext_timestamp(struct sched_cmd *cmd, enum zocl_ts_type ts)
+{
```
Comment (larry9523) [inline src/runtime_src/driver/zynq/drm/zocl/sched_exec.c]: Seems this function can be static

Comment (mamin506) [reply src/runtime_src/driver/zynq/drm/zocl/sched_exec.c]: There are a set of set_cmd_*() function should be static. How about let's clean up this file in 2019.2.

File: src/runtime_src/driver/zynq/drm/zocl/sched_exec.c [hunk 2]
```diff
@@ -755,6 +755,60 @@ set_cmd_state(struct sched_cmd *cmd, enum cmd_state state)
...
+/**
+ * set_cmd_ext_cu_idx() - Set external cu_idx of a command
+ *
+ * The cu_idx is reflected externally through the command packet
+ *
+ * @cmd: command object
+ * @cu_idx: CU idex
+ */
+inline void
+set_cmd_ext_cu_idx(struct sched_cmd *cmd, int cu_idx)
```
[the remark this answers is not in the document]
Comment (mamin506) [reply src/runtime_src/driver/zynq/drm/zocl/sched_exec.c]: The same as set_cmd_*() functions.

File: src/runtime_src/driver/zynq/drm/zocl/sched_exec.c [hunk 3]
```diff
@@ -755,6 +755,60 @@ set_cmd_state(struct sched_cmd *cmd, enum cmd_state state)
...
+		return;
+	/* Use the first 4 32bits in the regmap to record CU start/end time.
+	 * To let user application know the CU start/end time.
+	 * The first  32 bits - CU start seconds
+	 * The second 32 bits - CU start microseconds
+	 * The third  32 bits - CU done  seconds
+	 * The fourth 32 bits - CU done  microseconds
+	 * Use 32 bits timestamp is good enough for this purpose for now.
+	 */
+	do_gettimeofday(&tv);
```
[the remark this answers is not in the document]
Comment (mamin506) [reply src/runtime_src/driver/zynq/drm/zocl/sched_exec.c]: I think us is good enough. Do you think we really need ns resolution?

File: src/runtime_src/driver/zynq/drm/zocl/sched_exec.c [hunk 4]
```diff
@@ -1038,6 +1092,7 @@ mark_cmd_complete(struct sched_cmd *cmd)
 	if (zdev->ert || zdev->exec->polling_mode)
 		--cmd->sched->poll;
 	release_slot_idx(cmd->ddev, cmd->slot_idx);
+	set_cmd_ext_timestamp(cmd, CU_DONE_TIME);
```
[the remark this answers is not in the document]
Comment (mamin506) [reply src/runtime_src/driver/zynq/drm/zocl/sched_exec.c]: I thought that in before. But cmd is at outside of the scope of cu_done().
Actually, I should rewrite cu_done(). Let me make this change now.

File: src/runtime_src/driver/zynq/drm/zocl/sched_exec.c (hunk 1 above)
Comment (mamin506) [reply src/runtime_src/driver/zynq/drm/zocl/sched_exec.c]: See my latest commit.

**Pull request merged**